qiskit/transpiler/passes/layout/noise_adaptive_layout.py

```diff
@@ -71,6 +72,21 @@ def __init__(self, backend_prop):
         else:
             self.target = None
             self.backend_prop = backend_prop
+
+            if coupling_map:
+                # A backend might have more properties than qubits/gates in the configuration. This is a
+                # problem that the Target path should handle differently (by solving that possible
+                # inconsistency internally). For the non-target path, this is a possible solution.
+                # See https://github.com/Qiskit/qiskit/issues/7677
+                self.backend_prop.gates = filter(
+                    lambda ginfo: ginfo.gate == "cx"
+                    and tuple(ginfo.qubits) in coupling_map.graph.edge_list(),
```

Comment:
For performance it might make sense to collect the edge_list into a `set()`. Event though `edge_list()` returns a rust container it's a sequence type based on a rust Vec, so there is an `O(n^2)` overhead here because we're iterating over all edges for each edge in the backend.

Reply:
Good call! easy optimization 1e52ac8caff4082710fb5c548ea04d9f0823ef3a